### PR TITLE
Update to Rust 1.69.0

### PR DESCRIPTION
**Pull request by thomcc**
Not fully tested, but compiles. (I'll let CI do the testing)

### CODE REVIEW

**Comment by thomcc** [comment]
Depends on https://github.com/tcdi/postgrestd/pull/44 (although that one shouldn't directly get merged into 1.67.1 for obvious reasons)

File: plrustc/rust-toolchain.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [toolchain]
-channel = "1.67.1"
-components = [ "rustfmt", "rust-src", "rustc-dev", "cargo" ]
+channel = "1.69.0"
+components = [ "rustfmt", "rust-src", "rustc-dev", "cargo", "llvm-tools" ]
```
Comment (eeeebbbbrrrr) [inline plrustc/rust-toolchain.toml:3]: A++

I ran into needing that locally and wasn't sure what to do.  I ended up figuring out what I needed to hack from the CLI, but this is better.

**Review by eeeebbbbrrrr — APPROVED** [review]
I think this is fine.  If it's not a big hassle to teach `build.sh` to get the rust version from the `rust-toolchain.toml` file, I'd like to see that.

File: plrustc/build.sh [hunk 2]
```diff
@@ -25,8 +25,8 @@ fi
 export RUSTC_BOOTSTRAP=1
 
 version=$($RUSTC --version | cut -d ' ' -f 2)
-if [ "$version" != "1.67.1" ]; then
-    echo "rustc ('$RUSTC') is not version 1.67.1" >&2
+if [ "$version" != "1.69.0" ]; then
```
[the remark this answers is not in the document]
Comment (thomcc) [reply plrustc/build.sh:28]: Not easily, parsing toml files is hard and this is designed to catch the case that rustup lags behind or isn't being used.

File: plrustc/rust-toolchain.toml (hunk 1 above)
Comment (thomcc) [reply plrustc/rust-toolchain.toml:3]: It's kind of a rustc bug that it's separately needed, TBH.

Comment (eeeebbbbrrrr) [reply plrustc/rust-toolchain.toml:3]: okay.  I'll just anxiously await the plrust installer tool.

File: plrustc/build.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (eeeebbbbrrrr) [reply plrustc/build.sh:28]: okay.  I'll just anxiously await the plrust installer tool.

**Comment by eeeebbbbrrrr** [comment]
Ship it whenever CI goes green.  Oh, it just did!

**Pull request merged**